mycelo/genesis/genesis_state.go

```diff
@@ -608,87 +676,303 @@ func (ctx *deployContext) deployReserve() error {
+		{"StableToken", ctx.genesisConfig.StableToken},
+		{"StableTokenEUR", ctx.genesisConfig.StableTokenEUR},
+	}
+	for _, token := range tokens {
+		err := ctx.deployCoreContract(token.contract, func(contract *contract.EVMBackend) error {
+			return contract.SimpleCall("initialize",
+				token.cfg.Name,
+				token.cfg.Symbol,
+				token.cfg.Decimals,
+				env.MustProxyAddressFor("Registry"),
+				token.cfg.Rate.BigInt(),
+				token.cfg.InflationFactorUpdatePeriod,
+				token.cfg.InitialBalances.Accounts(),
+				token.cfg.InitialBalances.Amounts(),
+				token.cfg.ExchangeIdentifier,
+			)
+		})
+		if err != nil {
+			return err
+		}
+
+		stableTokenAddress := env.MustProxyAddressFor(token.contract)
+
+		if token.cfg.Frozen {
+			ctx.logger.Info("Freezing StableToken", "contract", token.contract)
+			err = ctx.contract("Freezer").SimpleCall("freeze", stableTokenAddress)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Configure StableToken Oracles
+		for _, oracleAddress := range token.cfg.Oracles {
+			ctx.logger.Info("Adding oracle for StableToken", "contract", token.contract, "oracle", oracleAddress)
+			err = ctx.contract("SortedOracles").SimpleCall("addOracle", stableTokenAddress, oracleAddress)
+			if err != nil {
+				return err
+			}
+		}
+
+		// If requested, fix goldPrice of stable token
+		if token.cfg.GoldPrice != nil {
+			ctx.logger.Info("Fixing StableToken goldPrice", "contract", token.contract)
+
+			// first check if the admin is an authorized oracle
+			authorized := false
+			for _, oracleAddress := range token.cfg.Oracles {
+				if oracleAddress == ctx.accounts.AdminAccount().Address {
+					authorized = true
+					break
+				}
+			}
+
+			if !authorized {
+				ctx.logger.Warn("Fixing StableToken goldprice requires setting admin as oracle", "admin", ctx.accounts.AdminAccount().Address)
+				err = ctx.contract("SortedOracles").SimpleCall("addOracle", stableTokenAddress, ctx.accounts.AdminAccount().Address)
+				if err != nil {
+					return err
+				}
+			}
+
+			ctx.logger.Info("Reporting price of StableToken to oracle", "contract", token.contract)
+			err = ctx.contract("SortedOracles").SimpleCall("report",
+				stableTokenAddress,
+				token.cfg.GoldPrice.BigInt(),
+				common.ZeroAddress,
+				common.ZeroAddress,
+			)
+			if err != nil {
+				return err
+			}
+
+			ctx.logger.Info("Add StableToken to the reserve", "contract", token.contract)
+			err = ctx.contract("Reserve").SimpleCall("addToken", stableTokenAddress)
+			if err != nil {
+				return err
+			}
+		}
+
+		ctx.logger.Info("Whitelisting StableToken as a fee currency", "contract", token.contract)
+		err = ctx.contract("FeeCurrencyWhitelist").SimpleCall("addToken", stableTokenAddress)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ctx *deployContext) createAccounts(accs []env.Account, namePrefix string) error {
+	accounts := ctx.contract("Accounts")
+
+	for i, acc := range accs {
+		name := fmt.Sprintf("%s %02d", namePrefix, i)
```

Comment:
maybe should be `%03d` or just `%d` for consistency if we ever use mycelo for generating genesis files for large testnets.

Reply:
good call